src/test/kotlin/org/jitsi/nlj/rtp/org/jitsi/nlj/rtp/bandwidthestimation/BandwidthEstimationTest.kt

```diff
@@ -42,26 +44,32 @@ abstract class FixedRateSender(
     abstract fun nextPacketSize(): Int
 
     fun schedulePacket(justSent: Boolean) {
-        if (!running || rate <= 0 || nextPacketSize() == 0)
+        if (!running || rate <= 0 || nextPacketSize() == 0) {
             nextPacket = null
-        else {
-            val interPacketTime = (rate * 1e9 / nextPacketSize()).toLong()
-            var packetDelayTime = interPacketTime
-
-            if (!justSent && lastSend != null)
-                packetDelayTime -= Duration.between(lastSend, clock.instant()).toNanos()
+        } else {
+            var packetDelayTime: Long
```

Comment:
you can combine the declaration with assignment here:
```
val packetDelayTime = lastSend?.let {
    packetDelayTime = (nextPacketSize() * 1e9 / rate).toLong()
    if (!justSent) {
        packetDelayTime -= Duration.between(lastSend, clock.instant()).toNanos()
    }
    packetDelayTime
} ?: run {
    0
}
```

Reply:
actually, looking more, i think your life might be easier if instead of making `lastSent` nullable, you make it non-null and use a `NEVER` value (i've done this in a couple places, you can check where it's defined).  then this could become:

```
val packetDelayTime = when(lastSent) {
    NEVER -> 0
    else -> {
        val delayTime = (nextPacketSize() * 1e9 / rate).toLong()
        if (!justSent) {
            delayTime -= Duration.between(lastSend, clock.instant()).toNanos()
        }
        delayTime
    }
}
```

Reply:
also, is there a way we can avoid the `* 1e9` thing?  can we have some helper to do it and name it in terms of a unit conversion?